packages/backend-plugin-api/src/services/definitions/AuditorService.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ErrorLike } from '@backstage/errors';
+import type { JsonObject } from '@backstage/types';
+import type { Request } from 'express';
+
+/** @public */
+export type AuditorEventStatus<E = ErrorLike> =
+  | { status: 'unknown' }
+  | { status: 'initiated' }
+  | { status: 'succeeded' }
+  | ({
+      status: 'failed';
+    } & ({ error: E } | { errors: E[] }));
+
+/**
+ * Options for creating an auditor event.
+ *
+ * @public
+ */
+export type AuditorEventOptions<T extends JsonObject> = {
+  /**
+   * Use kebab-case to name audit events (e.g., "user-login", "file-download").
+   *
+   * The `pluginId` already provides plugin/module context, so avoid redundant prefixes in the `eventId`.
+   */
+  eventId: string;
+
+  /** (Optional) The associated HTTP request, if applicable. */
+  request?: Request<any, any, any, any, any>;
+
+  /** (Optional) An identifier for the entity or user who triggered the event. */
+  actorId?: string;
+
+  /** (Optional) Additional metadata relevant to the event, structured as a JSON object. */
+  meta?: T;
+} & AuditorEventStatus;
+
+/** @public */
+export type AuditorEventLevels = keyof Omit<
+  AuditorService,
+  'createAuditorEvent'
+>;
+
+/** @public */
+export type AuditorCreateEvent<
+  T extends JsonObject,
+  E = ErrorLike,
+> = (options: {
+  /**
+   * Use kebab-case to name audit events (e.g., "user-login", "file-download").
+   *
+   * The `pluginId` already provides plugin/module context, so avoid redundant prefixes in the `eventId`.
+   */
+  eventId: string;
+
+  level?: Exclude<AuditorEventLevels, 'error'>;
+
+  /** (Optional) The associated HTTP request, if applicable. */
+  request?: Request<any, any, any, any, any>;
+
+  /** (Optional) An identifier for the entity or user who triggered the event. */
+  actorId?: string;
```

Comment:
Thinking it would be best to have to option to pass a `credentials` object here as an option, rather than a raw actor ID

Reply:
Just a thought: passing in credentials requires the `auth` service to be injected here to be able to do anything useful with those credentials right? That implies that the auth service won't be able to emit auditing events (due to circular dependencies). Maybe that's fine, but just noting.

Reply:
[This](https://github.com/backstage/backstage/pull/25060#discussion_r1670954649) was the original reason for keeping `actorId`. Thoughts?

Reply:
Ah, in that case we can switch to credentials, you always have access to the plugin's own credentials via https://backstage.io/docs/reference/backend-plugin-api.authservice.getownservicecredentials

There will already be a dep on the auth service btw Freben, since we're generally reading the credentials from the request option